services/121-service/src/payments/payments.service.ts

```diff
@@ -1270,4 +1273,54 @@ export class PaymentsService {
       `FinancialServiceProviderName ${financialServiceProviderName} not supported in fsp export`,
     );
   }
+
+  public async getTransactions({
+    programId,
+    payment,
+  }: {
+    programId: number;
+    payment: number;
+  }): Promise<TransactionResponseDto[]> {
+    const transactions =
+      await this.transactionScopedRepository.getTransactionsForPayment({
+        programId,
+        payment,
+      });
+    if (!transactions || transactions.length === 0) {
+      return [];
+    }
+
+    const fullnameNamingConvention = (
```

Comment:
Can we put the below logic into 1 query into the PaymentRepository instead of doing this less efficiently here?